libraries/AP_TopoLift/AP_TopoLiftSinglePointEstimate.cpp

```diff
@@ -0,0 +1,90 @@
+#include "AP_TopoLiftSinglePointEstimate.h"
+#include <AP_Terrain/AP_Terrain.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_Math/vector2.h>
+
+
+bool AP_TopoLiftSinglePointEstimate::get_terrain_height(const Location loc, float& height_amsl) {
+
+    AP_Terrain *terrain = AP::terrain();
+    if (terrain == nullptr) {
+        return false;
+    }
+
+    // status() handles enable check.
+    if (terrain->status() != AP_Terrain::TerrainStatusOK) {
+        return false;
+    }
+
+    const auto corrected = true;
+    return terrain->height_amsl(loc, height_amsl, corrected);
+
+
+    return true;
+}
+
+bool AP_TopoLiftSinglePointEstimate::get_terrain_grad(const Location loc, Vector2f& grad) {
+
+    AP_Terrain *terrain = AP::terrain();
+    if (terrain == nullptr) {
+        return false;
+    }
+
+    // status() handles enable check.
+    if (terrain->status() != AP_Terrain::TerrainStatusOK) {
+        return false;
+    }
+
+    return terrain->gradient(loc, grad);
+}
+
+bool AP_TopoLiftSinglePointEstimate::get_wind_at_current_loc(Vector3f& wind) {
+    const AP_AHRS &ahrs = AP::ahrs();
+
+    return ahrs.wind_estimate(wind);
```

Comment:
Should we *really* be using `AP::ahrs()` here?  Your simulation shouldn't really depend on what NavEKF3 is up to - if your fix is wandering your simulation is going to follow that wandering vehicle rather than represent the true lift value the simulated vehicle should be experiencing.

I'm hoping to have a set of AHRS backend results available from the SITL backend which might allow for using the "perfect" AHRS for determining the simulatedwind and a stnadrd backend's results when using on a real vehicle.